plan.go

```diff
@@ -2837,17 +2839,20 @@ func (r *emptyFieldsPlan) do(ctx *execCtx, f func(id interface{}, data []interfa
 
 type wrapFilterPlan struct {
 	*filterDefaultPlan
+	once  sync.Once
```

Comment:
Something, with proper error handling and synchronization added, like discussed above

```go
// error handling omitted
s, _ := db.Prepare("select * from t where exists (select * from t where x == ?)")
go s.Exec(1)
go s.Exec(4)
```

In the setup of the test prepare the table, fill in some data, let the test run for few seconds so the race has a better chance to be detected.

The query is just an example, use anything you like that uses the new codepath with the new map field in execCtx.